plugins/pidfile.c

```diff
@@ -88,16 +89,28 @@ static void pidfile_callback(void *UNUSED(arg), int fd, int UNUSED(events))
 static void pidfile_reconf(void *_null)
 {
 	static char name[MAX_ARG_LEN];
-
 	svc_t *svc;
 	(void)(_null);
+	int restart = 0;
+
+	do {
+		restart = 0;
 
-	for (svc = svc_iterator(1); svc; svc = svc_iterator(0)) {
-		if (svc->state == SVC_RUNNING_STATE && !svc_is_changed(svc) && !svc_is_starting(svc)) {
+		for (svc = svc_iterator(1); svc; svc = svc_iterator(0)) {
 			snprintf(name, MAX_ARG_LEN, "svc%s", svc->cmd);
-			cond_set_path(cond_path(name), COND_ON);
+			if (svc->state == SVC_RUNNING_STATE &&
+			    !svc_is_changed(svc) &&
```

Comment:
On second thought, that is actually correct coding style.  The Emacs linux-mode usually does the right thing :)